include/kframework/lemmas/int-simplification.k

```diff
@@ -158,4 +161,26 @@ module INT-SIMPLIFICATION-COMMON
 
     rule A -Int B +Int C <=Int D => false requires D <Int A -Int B andBool 0 <=Int C [simplification]
 
+  // ###########################################################################
+  // up/Int
+  // ###########################################################################
+
+    rule [upInt-lt-true]:
+      ((X up/Int Y) *Int Y) <Int Z => true
+      requires X <=Int (Z -Int Y) andBool 0 <Int Y
+      [simplification]
+
+    rule [upInt-lt-false]:
+      ((X up/Int Y) *Int Y) <Int X => false
+      requires 0 <Int Y
+      [simplification]
+
+    rule [lengthBytes-upInt-32-lower-bound]:
+      lengthBytes(X) <=Int notMaxUInt5 &Int lengthBytes(X) +Int 31 => true
+      [simplification, smt-lemma]
+
+    rule [lengthBytes-upInt-32-upper-bound]:
+      lengthBytes(X) +Int 32 >Int notMaxUInt5 &Int lengthBytes(X) +Int 31 => true
+      [simplification, smt-lemma]
```

Comment:
Not sure these lemmas belong here, and they seem pretty specific. Could we replace `lengthBytes(X)` with just a variable `Y #And Y >=Int 0`, and then these lemmas just become about comparisons and bitwise operations (so moved to `bitwise-simplification.k`)?

Reply:
So, `notMaxUInt5 &Int lengthBytes(X) +Int 31` is `lengthBytes(X) up/Int 32`, that's why I put them here instead of there, and these have appeared in that form in Optimism when copying some data in memory, so they are likely to reappear. 

We could also make the `up/Int` explicit with a simplification and then reason about it instead? Possibly more generally, also, if the second operand is a power of two...